### PR TITLE
Add cmake options to disable optional dependencies

### PR DESCRIPTION
**Pull request by linkmauve**
This allows people to build without some of these dependencies even though they are present on the system. This is helpful especially when building a package for other people.

### CODE REVIEW

**Comment by ligfx** [comment]
Should hires textures ever be disabled? That seems like one of the big selling points of Dolphin. Maybe an option to force building the local copy would be better.

**Comment by linkmauve** [comment]
I guess most people never once tried this feature, nor know anything about it, or don’t even have a computer powerful enough to use it.

But the point of this PR is to make optional dependencies actually optional at build time, so the person building Dolphin doesn’t need to manually remove them from their system in order not to use them.

(And my reason for wanting to disable SOIL is to make the nogui frontend build properly even with `-DENABLE_EGL=TRUE`, which doesn’t pull libGL.so in, which SOIL depends on.)

**Comment by degasus** [comment]
Hm, I dislike that the GUI still shows the custom texture option, and we still try to load them but fail. So the usability with disabled SOIL is terrible.

But as we should replace SOIL with libpng itself, we hopefully get rid of this at somepoint. so LGTM

**Comment by linkmauve** [comment]
I just dropped the SOIL part from this PR, I will fix that in another PR, this one only does the audio backends, LLVM and Bluetooth now.

File: CMakeLists.txt [hunk 1]
```diff
@@ -445,62 +451,84 @@ if (OPENGL_GL)
...
+if(ENABLE_ALSA)
+	include(FindALSA OPTIONAL)
+	if(ALSA_FOUND)
+		add_definitions(-DHAVE_ALSA=1)
+		message("ALSA found, enabling ALSA sound backend")
+	else()
+		add_definitions(-DHAVE_ALSA=0)
+		message("ALSA NOT found, disabling ALSA sound backend")
+	endif(ALSA_FOUND)
+else(ENABLE_ALSA)
```
Comment (Orphis) [inline CMakeLists.txt]: Modern CMake styleguide says that else and endif() shouldn't have the condition from the if copied.

Comment (linkmauve) [inline CMakeLists.txt]: Fixed, I just wanted to adhere to the existing code style, but I fixed it to match this instead.

**Pull request merged**